api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -86,6 +86,12 @@ public interface RewriteDataFiles
 
   int MAX_CONCURRENT_FILE_GROUP_REWRITES_DEFAULT = 5;
 
+  /**
+   * The max number of files to be rewritten. By default, this value will be 0 (To rewrite all the
+   * files)
+   */
+  String MAX_FILES_TO_REWRITE = "max-files-to-rewrite";
```

Comment:
I'm not a fan of magic 0's. Could we just not-limit the number of files if this value isn't set? (0 could also just be an unacceptable value for a user to pass in as well)

Reply:
Thank you for the feedback @RussellSpitzer  !  Sure I can cap it to a high number such as INT_MAX (or even LONG_MAX which I dont think is a bit too much)  and update documentation ?